File: scripts/make_sample_xlsx.py
#!/usr/bin/env python3
"""Genera examples/sample.xlsx a partir de examples/sample.csv usando openpyxl."""
import csv
import argparse
from openpyxl import Workbook


def csv_to_xlsx(csv_path, xlsx_path):
    wb = Workbook()
    ws = wb.active
    with open(csv_path, newline='', encoding='utf-8') as f:
        reader = csv.reader(f)
        for r, row in enumerate(reader, start=1):
            for c, val in enumerate(row, start=1):
                ws.cell(row=r, column=c, value=val)
    wb.save(xlsx_path)


def main():
    parser = argparse.ArgumentParser(description='Generar sample.xlsx desde sample.csv')
    parser.add_argument('--csv', default='examples/sample.csv')
    parser.add_argument('--out', default='examples/sample.xlsx')
    args = parser.parse_args()
    csv_to_xlsx(args.csv, args.out)
    print(f'Generado: {args.out}')


if __name__ == '__main__':
    main()

File: scripts/make_sample_xlsx.js
#!/usr/bin/env node
/*
Genera examples/sample.xlsx a partir de examples/sample.csv usando la librería 'xlsx' de Node.
Uso:
  npm install xlsx
  node scripts/make_sample_xlsx.js --out examples/sample.xlsx
Opciones:
  --csv <ruta>   Ruta al CSV de entrada (por defecto: examples/sample.csv)
  --out <ruta>   Ruta al xlsx de salida (por defecto: examples/sample.xlsx)
*/
const fs = require('fs');
const path = require('path');
const XLSX = require('xlsx');

const argv = process.argv.slice(2);
let csvPath = 'examples/sample.csv';
let outPath = 'examples/sample.xlsx';
for (let i = 0; i < argv.length; i++) {
  if (argv[i] === '--csv' && argv[i + 1]) { csvPath = argv[i + 1]; i++; }
  if (argv[i] === '--out' && argv[i + 1]) { outPath = argv[i + 1]; i++; }
}

if (!fs.existsSync(csvPath)) {
  console.error(`CSV no encontrado: ${csvPath}`);
  process.exit(1);
}

const csv = fs.readFileSync(csvPath, 'utf8');
const rows = csv.split(/\r?\n/).filter(r => r.length > 0).map(r => r.split(','));
const ws = XLSX.utils.aoa_to_sheet(rows);
const wb = XLSX.utils.book_new();
XLSX.utils.book_append_sheet(wb, ws, 'Sheet1');
XLSX.writeFile(wb, outPath);
console.log(`Generado: ${outPath}`);
